app/scripts/directives/nav.js

```diff
@@ -242,6 +243,26 @@ angular.module('openshiftConsole')
           _.set($rootScope, 'nav.showMobileNav', visible);
         };
 
+        var catalogPath = "/catalog";
+        var projectPath = "/project";
+        var projectsPath = "/projects";
+
+        var setContextPickerSelectedOption = function(selector) {
+          var path = $location.path();
+          if (path === catalogPath) {
+            selector.selectpicker('val', "catalog");
+          }
+          if (_.includes(path, projectPath)) {
```

Comment:
`_.startsWith` is probably more accurate. This could be an `else if` as well.

What happens on pages without a project such as the about page? I think it makes sense to always set it to "application-console" if it's not catalog.